setups/validator/systemd/README.md

```diff
@@ -0,0 +1,159 @@
+
+Ensure the following ports are open in your firewall:
+
+| Port     | Reachability     | Purpose                           |
+| -------- | ---------------- | --------------------------------- |
+| TCP/8080 | inbound          | protocol/transaction interface    |
+| TCP/8081 | inbound/outbound | primary interface                 |
+| UDP/8084 | inbound/outbound | peer to peer state sync interface |
+| TCP/8443 | outbound         | metrics pushing                   |
+| TCP/9184 | localhost        | metrics scraping                  |
+
+> **Note**: Maybe you already noticed that port 8081 is using TCP, which conflicts with docs.iota.org and validator.info as well. This is a known bug, but the node is actually communicating via TCP.
+
+## Environment Variables (Optional)
+
+These environment variables are pre-configured for the testnet. You typically don't need to modify them unless you're using custom endpoints.
+
+```bash
+export IOTA_API_ENDPOINT="https://api.testnet.iota.cafe"
+export IOTA_FAUCET_ENDPOINT="https://faucet.testnet.iota.cafe/v1/gas"
+export IOTA_TOOLS_DOCKER_IMAGE="iotaledger/iota-tools:testnet"
+```
+
+## Setup Steps
+
+### 1. Download Configuration Template
+
+Download the validator configuration file template:
+
+```bash
+curl -L -o validator.yaml https://github.com/iotaledger/iota/raw/refs/heads/develop/nre/config/validator.yaml
+```
+
+### 2. Configure P2P Settings
+
+Configure the P2P settings in your `validator.yaml` file by following these steps:
+
+1. Update the external address:
+
+```yaml
+p2p-config:
+  external-address: /dns/<YOUR-DOMAIN>/udp/8084
+```
+
+> **Note**: Replace `<YOUR-DOMAIN>` with your validator's public domain name. This address must be accessible from the internet.
+
+2. Add the seed peers configuration:
+
+```yaml
+p2p-config:
+  listen-address: "0.0.0.0:8084"
+  external-address: /dns/<YOUR-DOMAIN>/udp/8084
+  anemo-config:
+    max-concurrent-connections: 0
+  seed-peers:
+    - address: /dns/access-0.r.testnet.iota.cafe/udp/8084
+      peer-id: 46064108d0b689ed89d1f44153e532bb101ce8f8ca3a3d01ab991d4dea122cfc
+    - address: /dns/access-1.r.testnet.iota.cafe/udp/8084
+      peer-id: 8ffd25fa4e86c30c3f8da7092695e8a103462d7a213b815d77d6da7f0a2a52f5
+```
+
+### 3. Configure Metrics Pushing Target
+
+Configure the metrics settings in your `validator.yaml` file by following these steps:
+
+Update the `push-url`:
+
+```yaml
+metrics:
+  push-interval-seconds: 60
+  push-url: https://metrics-proxy.testnet.iota.cafe:8443/publish/metrics
+```
+
+### 4. Download Genesis Block
+
+```bash
+curl -fLJO https://dbfiles.testnet.iota.cafe/genesis.blob
+```
+
+> **Note**: The URL is for the IOTA Testnet only.
+
+### 5. Make validator.info and Generate Validator Keys
+
+Generate the necessary key pairs for your validator, the key pairs will be stored in `key-pairs` folder.
+
+```bash
+./generate_validator_info.sh
+```
+
+After running `./generate_validator_info.sh`, you'll receive output similar to this:
```

Comment:
All the mentioned scripts don't exist. We need to rewrite that.

Reply:
It's in `setups/validator`.

Reply:
The script in there is only for docker. We need to create the same scripts for systemd.